### PR TITLE
added business-logic

### PR DESCRIPTION
**Pull request by eeveesharp**

### CODE REVIEW

File: Shop/Shop/Controllers/ProductController.cs [hunk 1]
```diff
@@ -39,27 +46,58 @@ public IActionResult Delete(int id)
...
 
-            return resultProduct;
+            return Ok(resultProduct);
         }
 
         [HttpPost]
-        public Product Create(Product product)
+        public IActionResult Create(Product product)
         {
+            if (!IsCorrectData(product))
```
Comment (Blackolf931) [inline Shop/Shop/Controllers/ProductController.cs:68]: You check Id on post method?
We don't have id field when we create instance

File: Shop/Shop/Controllers/ProductController.cs [hunk 2]
```diff
@@ -19,16 +21,21 @@ public class ProductController : ControllerBase
         }
 
         [HttpGet]
-        public List<Product> GetAll()
+        public IActionResult GetAll()
```
Comment (Blackolf931) [inline Shop/Shop/Controllers/ProductController.cs:24]: Why you change return type?

File: Shop/Shop/Controllers/ProductController.cs [hunk 3]
```diff
@@ -19,16 +21,21 @@ public class ProductController : ControllerBase
...
-            return resultProduct;
+            return Ok(resultProduct);
         }
 
         [HttpDelete("{id}")]
         public IActionResult Delete(int id)
         {
+            if (!IsCorrectId(id))
+            {
+                return BadRequest("Error");
```
Comment (Blackolf931) [inline Shop/Shop/Controllers/ProductController.cs:36]: Maybe throw exception?

**Review by Blackolf931 — CHANGES_REQUESTED** [review]

File: Shop/Shop/Controllers/ProductController.cs [hunk 4]
```diff
@@ -27,20 +29,33 @@ public List<Product> GetAll()
         }
 
         [HttpDelete("{id}")]
-        public IActionResult Delete(int id)
+        public Product Delete(int id)
         {
+            if (IsCorrectId(id) is null)
```
Comment (hromau) [inline Shop/Shop/Controllers/ProductController.cs:34]: code is hard to read. there is should be one check, like if(IsCorrectId(id)) { ... } without additional check on null

File: Shop/Shop/Controllers/ProductController.cs [hunk 5]
```diff
@@ -61,5 +83,26 @@ public Product Create(Product product)
 
             return resultProduct;
         }
+
+        private bool IsCorrectQuantity(Product product)
```
Comment (hromau) [inline Shop/Shop/Controllers/ProductController.cs:87]: it is class Controller. methods bellow not related to controller actions. move it to somewhere. look at SOLID. this case is first letter S.

**Review by hromau — CHANGES_REQUESTED** [review]

**Review by hromau — CHANGES_REQUESTED** [review]

**Review by hromau — APPROVED** [review]

**Review by Blackolf931 — APPROVED** [review]

**Pull request merged**